src/main/io/vtx_tramp.c

```diff
@@ -0,0 +1,534 @@
+static void trampCmsUpdateStatusString(void); // Forward
+#endif
+
+static void trampWriteBuf(uint8_t *buf)
+{
+    serialWriteBuf(trampSerialPort, buf, 16);
+}
+
+static uint8_t trampChecksum(uint8_t *trampBuf)
+{
+    uint8_t cksum = 0;
+
+    for (int i = 1 ; i < 14 ; i++)
+        cksum += trampBuf[i];
+
+    return cksum;
+}
+
+void trampCmdU16(uint8_t cmd, uint16_t param)
+{
+    if (!trampSerialPort)
+        return;
+
+    memset(trampReqBuffer, 0, ARRAYLEN(trampReqBuffer));
+    trampReqBuffer[0] = 15;
+    trampReqBuffer[1] = cmd;
+    trampReqBuffer[2] = param & 0xff;
+    trampReqBuffer[3] = (param >> 8) & 0xff;
+    trampReqBuffer[14] = trampChecksum(trampReqBuffer);
+    trampWriteBuf(trampReqBuffer);
+}
+
+void trampSetFreq(uint16_t freq)
+{
+    trampConfFreq = freq;
+}
+
+void trampSendFreq(uint16_t freq)
+{
+    trampCmdU16('F', freq);
+}
+
+void trampSetBandChan(uint8_t band, uint8_t chan)
+{
+    trampSetFreq(vtx58FreqTable[band - 1][chan - 1]);
+}
+
+void trampSetRFPower(uint16_t level)
+{
+    trampConfPower = level;
+}
+
+void trampSendRFPower(uint16_t level)
+{
+    trampCmdU16('P', level);
+}
+
+// return false if error
+bool trampCommitChanges()
+{
+    if(trampStatus != TRAMP_STATUS_ONLINE)
+        return false;
+
+    trampStatus = TRAMP_STATUS_SET_FREQ_PW;
+    return true;
+}
+
+void trampSetPitmode(uint8_t onoff)
+{
+    trampCmdU16('I', onoff ? 0 : 1);
+}
+
+static uint8_t trampPendingQuery = 0; // XXX Assume no code/resp == 0
+
+void trampQuery(uint8_t cmd)
+{
+    trampPendingQuery = cmd;
+    trampCmdU16(cmd, 0);
+}
+
+void trampQueryR(void)
+{
+    trampQuery('r');
+}
+
+void trampQueryV(void)
+{
+    trampQuery('v');
+}
+
+void trampQueryS(void)
+{
+    trampQuery('s');
+}
+
+#define TRAMP_SERIAL_OPTIONS (SERIAL_BIDIR)
+
+bool trampInit()
+{
+    serialPortConfig_t *portConfig = findSerialPortConfig(FUNCTION_VTX_CONTROL);
+
+    if (portConfig) {
+        trampSerialPort = openSerialPort(portConfig->identifier, FUNCTION_VTX_CONTROL, NULL, 9600, MODE_RXTX, TRAMP_SERIAL_OPTIONS);
+    }
+
+    if (!trampSerialPort) {
+        return false;
+    }
+
+    return true;
+}
+
+void trampHandleResponse(void)
+{
+    uint8_t respCode = trampRespBuffer[1];
+
+    switch (respCode) {
+    case 'r':
+        trampRFFreqMin = trampRespBuffer[2]|(trampRespBuffer[3] << 8);
+        trampRFFreqMax = trampRespBuffer[4]|(trampRespBuffer[5] << 8);
+        trampRFPowerMax = trampRespBuffer[6]|(trampRespBuffer[7] << 8);
+        trampStatus = TRAMP_STATUS_ONLINE;
+        break;
+
+    case 'v':
+        {
+            trampCurFreq = trampRespBuffer[2]|(trampRespBuffer[3] << 8);
+            trampCurConfigPower = trampRespBuffer[4]|(trampRespBuffer[5] << 8);
+            trampCurPower = trampRespBuffer[8]|(trampRespBuffer[9] << 8);
+            vtx58_Freq2Bandchan(trampCurFreq, &trampCurBand, &trampCurChan);
+            trampCurPitmode = trampRespBuffer[7];
+
+            if (trampStatus == TRAMP_STATUS_CHECK_FREQ_PW)
```

Comment:
this state transition is hidden quite deep and far from state automaton using it ...